backend/src/db/pagination.rs

```diff
@@ -0,0 +1,160 @@
+    offset: i32,
+}
+
+/// A trait intended for enabling pagination in diesels query builder.
+pub trait Paginate: Sized {
+    /// Return a paginated version of a query for a specific page number.
+    fn paginate(self, page: Option<i32>) -> PaginatedQuery<Self>;
+}
+
+impl<T> Paginate for T {
+    fn paginate(self, page: Option<i32>) -> PaginatedQuery<Self> {
+        // allow optional pages and disallow non positive pages
+        let actual_page = max(page.unwrap_or(MIN_PAGE), MIN_PAGE);
+        PaginatedQuery {
+            query: self,
+            per_page: DEFAULT_PER_PAGE,
+            page: actual_page,
+            offset: (actual_page - 1) * DEFAULT_PER_PAGE,
+        }
+    }
+}
+
+impl<T> PaginatedQuery<T> {
+    /// Set the number of rows returned by the query.
+    #[must_use]
+    #[allow(clippy::missing_const_for_fn)]
+    pub fn per_page(self, per_page: Option<i32>) -> Self {
+        // allow optional per_page and disallow non positive per_page
+        let actual_per_page = max(per_page.unwrap_or(DEFAULT_PER_PAGE), MIN_PER_PAGE);
+        Self {
+            per_page: actual_per_page,
+            offset: (self.page - 1) * actual_per_page,
+            ..self
+        }
+    }
+
+    /// Execute the query returning one [`Page`] of rows.
+    ///
+    /// # Errors
+    /// Unknown, diesel doesn't say why it might error.
+    pub async fn load_page<'query, 'conn, U>(
+        self,
+        conn: &'conn mut AsyncPgConnection,
+    ) -> QueryResult<Page<U>>
+    where
+        T: Send,
+        U: Send,
+        Self: LoadQuery<'query, AsyncPgConnection, (U, i64)> + 'query,
+    {
+        let page = self.page;
+        let per_page = self.per_page;
+        let query_result = self.load::<(U, i64)>(conn).await?;
+        let total = query_result.get(0).map_or(0, |x| x.1);
+        let results = query_result.into_iter().map(|x| x.0).collect();
+        #[allow(
+            clippy::cast_possible_truncation,
+            clippy::cast_precision_loss,
+            clippy::cast_lossless
+        )]
+        let total_pages = (total as f64 / per_page as f64).ceil() as i32;
+        Ok(Page {
+            results,
+            page,
+            per_page,
+            total_pages,
+        })
+    }
+}
+
+impl<T: Query> Query for PaginatedQuery<T> {
+    type SqlType = (T::SqlType, BigInt);
+}
+
+impl<T> QueryFragment<Pg> for PaginatedQuery<T>
+where
+    T: QueryFragment<Pg>,
+{
+    fn walk_ast<'b>(&'b self, mut out: AstPass<'_, 'b, Pg>) -> QueryResult<()> {
+        out.push_sql("SELECT *, COUNT(*) OVER () FROM (");
+        self.query.walk_ast(out.reborrow())?;
+        out.push_sql(") t LIMIT ");
+        out.push_bind_param::<Integer, _>(&self.per_page)?;
+        out.push_sql(" OFFSET ");
+        out.push_bind_param::<Integer, _>(&self.offset)?;
+        Ok(())
+    }
+}
+
+/// A page of results returned from a list endpoint.
+#[typeshare]
+#[derive(Debug, Serialize, PartialEq, Eq, Deserialize, ToSchema)]
```

Comment:
For utoipa to work with generics you need to add `#[aliases(PageSeedDto= Page<SeedDto>)]` here.
Afterwardes you need to add `PageSeedDto` as a component in api_doc.rs instead of `Page<SeedDto>`.

You have to define similar aliases for all generics you want to use with utoipa.

See [here](https://docs.rs/utoipa/latest/utoipa/derive.ToSchema.html#generic-schemas-with-aliases) for reference.